ember-style-guides.md

```diff
@@ -1,269 +1,312 @@
+- Use a consistent structure for components.
 
-- Same as above for Adapters.;
+```javascript
+export default Component.extend({
+  // service declarations
 
-- RTFM: guides.emberjavascript.com;
+  // native component properties
+  tagName: 'nav',
 
+  // passed properties
+  attrs: {},
 
-##Controllers
+  // custom properties
+  foo: 'foo',
 
-Controllers in general should not be used. Most controllers should will like this:
+  // lifecycle hooks in order of execution
+  init() {},
+  didReceiveAttrs() {},
+  didInsertElement() {},
 
+  // computed properties
+  fancyFoo: computed('foo', function() {
+    // body omitted
+  }),
 
-```javascript
-import Ember from 'ember';
+  // native functions
 
-const { Controller } = Ember;
+  // actions separated by a newline
+  actions: {
+    doFoo() {
+      // body omitted
+    },
 
-export default Controller.extend({
-attrs: {}
+    doBar() {
+      // body omitted
+    }
+  }
+});
+```
+
+- Set tag name. Components don’t need to be wrapped in a div as they are by default. Use the `tagName` property to modify a component before wrapping it.
+
+```javascript
+// good
+export default Component.extend({
+  tagName: 'nav'
 });
 ```
 
-We don’t do much more here. It is ok to sometimes add some code, for example a small computed property to filter an item. Another exception is obviously queryParams.
+- Don't over nest your components. Consider if your nesting adds better details or obfuscation for new developers. Heavy DOM nesting negatively impacts performance on mobile devices.
 
-Sometimes we still use Controllers in mobile where the saved nesting
-noticeably improves performance.
+- Align properties in Handlebars.
 
-##Routes
+```hbs
+# good
+{{component-name
+  property='foo'
+  action=(action 'actionName')}}
+```
 
-- Instead of setting a model directly on a controller (e.g. controller.modelName), use an attrs hook. This also makes it easy to distinguish items that _are_ on the controller proper.
+## Models
 
-An example route:
+- Use a consistent structure for models.
 
 ```javascript
-export default Route.extend({
-  model() {
-    return this.store.find(‘contact’);
-  },
+export default Model.extend({
+  // service declarations
 
-  setupController(controller, model) {
-    controller.set(‘attrs.contact’, model);
-  }
-});
-```
+  // attributes
+  title: attr('string'),
+  body: attr('string'),
 
-And in your templates, you can access attrs.contact. 
+  // relationships
+  comments: hasMany('comment'),
+  author: belongsTo('user'),
 
-- For multi model routes, leverage RSVP.hash:
-```javascript
-export default Route.extend({
-  model() {
-    return RSVP.hash({
-      followers: this.store.find(‘follower’),
-      following: this.store.find(‘following’)
-    }),
-  },
+  // computed properties
+  fancyTitle: computed('title', function() {
+    // return something
+  });
 
-  setupController(controller, model) {
-    controller.set(‘attrs’, model);
-  }   
+  // custom functions
```

Comment:
The model structure example is not too bad. Just need the service declaration example again and the custom functions.